app/src/main/java/de/westnordost/streetcomplete/quests/max_weight/AddMaxWeightForm.kt

```diff
@@ -0,0 +1,175 @@
+        poundInput = contentView.findViewById(R.id.poundInput)
+
+        val onTextChangedListener = TextChangedWatcher { checkIsFormComplete() }
+        tonInput?.addTextChangedListener(onTextChangedListener)
+        poundInput?.addTextChangedListener(onTextChangedListener)
+
+        tonInputSign = contentView.findViewById(R.id.tonInputSign)
+        poundInputSign = contentView.findViewById(R.id.poundInputSign)
+
+        weightUnitSelect = contentView.findViewById(R.id.weightUnitSelect)
+        val measurementUnits = countryInfo.measurementSystem
+        weightUnitSelect?.visibility = if (measurementUnits.size == 1) View.GONE else View.VISIBLE
+        weightUnitSelect?.adapter = ArrayAdapter(context!!, R.layout.spinner_item_centered, getSpinnerItems(measurementUnits))
+        weightUnitSelect?.setSelection(measurementUnits.size - 1) // prefer metric
+        weightUnitSelect?.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(parentView: AdapterView<*>, selectedItemView: View?, position: Int, id: Long) {
+                val heightUnit = if (weightUnitSelect?.selectedItem == "t") METRIC else IMPERIAL
+                switchLayout(heightUnit)
+            }
+
+            override fun onNothingSelected(parentView: AdapterView<*>) {}
+        }
+
+        /* Workaround for an Android bug that it assumes the decimal separator to always be the "."
+           for EditTexts with inputType "numberDecimal", independent of Locale. See
+           https://issuetracker.google.com/issues/36907764 .
+
+           Affected Android versions are all versions till (exclusive) Android Oreo. */
+
+        /* actually, let's not care about which separator the user uses, he might be confused
+           whether he should use the one as displayed on the sign or in his phone's locale */
+        //char separator = DecimalFormatSymbols.getInstance(getCountryInfo().getLocale()).getDecimalSeparator();
+        tonInput?.keyListener = DigitsKeyListener.getInstance("0123456789,.")
+
+        switchLayout(unit)
+    }
+
+    private fun switchLayout(unit: Measurement) {
+        val isMetric = unit == METRIC
+        val isImperial = unit == IMPERIAL
+
+        tonInputSign?.visibility = if (isMetric) View.VISIBLE else View.GONE
+        poundInputSign?.visibility = if (isImperial) View.VISIBLE else View.GONE
+
+        if (isMetric) tonInput?.requestFocus()
+        if (isImperial) poundInput?.requestFocus()
+    }
+
+    private fun getSpinnerItems(units: List<String>) = units.mapNotNull {
+        when(it) {
+            "metric" -> "t"
+            "imperial" -> "lbs"
+            else -> null
+        }
+    }
+
+    override fun onClickOk() {
+        if (userSelectedUnrealisticHeight()) {
+            confirmUnusualInput { applyMaxWeightFormAnswer() }
+        } else {
+            applyMaxWeightFormAnswer()
+        }
+    }
+
+    private fun userSelectedUnrealisticHeight(): Boolean {
```

Comment:
*weight

Best to Ctrl+f through this class if any "Height" is left from the copypasta.

Reply:
Fixed